ql/instruments/creditdefaultswap.cpp

```diff
@@ -414,6 +431,17 @@ namespace QuantLib {
             ->accrualEndDate();
     }
 
+    const ext::shared_ptr<CashFlow>& CreditDefaultSwap::accrualRebate() const {
```

Comment:
Should you return a SimpleCashFlow here instead of just CashFlow? Doesn't make it very easy to use otherwise.

Reply:
Without thinking very much about it, I just returned the member as it was, i.e. `accrualRebate_` was a `ext::shared_ptr<CashFlow>` so I just returned it. I think it makes sense, as you say, to change `upfrontPayment_` and `accrualRebate_` to `ext::shared_ptr<CashFlow>` for clarity i.e. they will never be anything else. I will do that now. Thanks.